src/main/java/uk/gov/hmcts/darts/common/repository/ExternalObjectDirectoryRepository.java

```diff
@@ -367,9 +368,32 @@ boolean hasMediaNotBeenCopiedFromInboundStorage(MediaEntity media, ObjectRecordS
     )
     List<ExternalObjectDirectoryEntity> findAllByStatusAndManifestFile(ObjectRecordStatusEntity status, String manifestFile);
 
+    @Query(
+        value = """
+         SELECT od.eod_id FROM darts.external_object_directory od WHERE od.med_id IN
+            (
+                SELECT med_id FROM darts.external_object_directory
+                WHERE elt_id=1 AND ors_id=2
+            ) AND elt_id=3 AND ors_id=2 
+            AND (floor (extract (epoch from (current_timestamp::timestamp - last_modified_ts::timestamp)) / 60 / 60) >= :hoursInArm)
+        """, nativeQuery = true
+    )
+    List<Integer> findAllInboundArmMediaExceedingHours(Pageable resultLimit, Integer hoursInArm);
```

Comment:
i think it's better to reuse the method that a few of the other jobs are already using:-
findMediaFileIdsIn2StorageLocationsBeforeTime

I don't see a batch size mentioned in the ticket. is this a requirement?

See InboundAudioDeleterProcessorImpl as an example

Reply:
nah I was told to use pagination as opposed to batch size

Reply:
my question is why we are limiting the result at all?
who told you to use pagination?

Reply:
There is no requirement of using a batch or pagesize in the ticket.

Reply:
in mind mind it was always good to have the flexibility of pagination. if null is passed through pagination is disabled. In that way it adds flexibility to the underlying database call and fits in with the underlying concept of the task. I will remove it.